### PR TITLE
Feature/new tethering algorithm

### PR DESCRIPTION
**Pull request by SmischenkoB**

### CODE REVIEW

File: Ship_Game/Ships/Ship.cs [hunk 1]
```diff
@@ -436,9 +436,9 @@ float GetYBankAmount(FixedSimTime timeStep)
             }
         }
 
-        public bool IsPlatformOrStation => ShipData.IsPlatformOrStation;
-        public bool IsShipyard => ShipData.IsShipyard;
-        public bool IsStation => ShipData.IsStation;
+        public bool IsPlatformOrStation => ShipData.IsPlatformOrStation || MaxFTLSpeed == 0 || MaxSTLSpeed == 0 ;
+        public bool IsShipyard => ShipData.IsShipyard || MaxFTLSpeed == 0 || MaxSTLSpeed == 0;
+        public bool IsStation => ShipData.IsStation || MaxFTLSpeed == 0 || MaxSTLSpeed == 0;
```
Comment (gkapulis) [inline Ship_Game/Ships/Ship.cs:441]: This is not needed. In game all of this is ok without the speed. 
The only place it needs to be checked is the CreateStartingShips() edge case there a prototype might be a station and therefore its roledata is prototype.

File: Ship_Game/Ships/Ship_Initialize.cs [hunk 2]
```diff
@@ -332,7 +332,8 @@ public static Ship CreateShipNearPlanet(UniverseState us, string shipName, Empir
             Ship ship = CreateShipAt(us, shipName, owner, p, p.Position.GenerateRandomPointOnCircle(randomRadius, owner.Random), doOrbit);
             if (initLaunch && ship != null && !ship.IsPlatformOrStation)
                 ship.InitLaunch(LaunchPlan.Planet, p);
-
+            else if (initLaunch && ship != null && (ship.MaxFTLSpeed == 0 || ship.MaxSTLSpeed == 0))
+                ship.TetherToPlanet(p);
```
Comment (gkapulis) [inline Ship_Game/Ships/Ship_Initialize.cs:336]: Fix should be only in void CreateStartingShips()

**Review by gkapulis — CHANGES_REQUESTED** [review]

File: Ship_Game/Ships/Ship_Initialize.cs [hunk 3]
```diff
@@ -330,9 +330,13 @@ public static Ship CreateShipNearPlanet(UniverseState us, string shipName, Empir
         {
             float randomRadius = owner.Random.Float(p.Radius - 100, p.Radius + 100);
             Ship ship = CreateShipAt(us, shipName, owner, p, p.Position.GenerateRandomPointOnCircle(randomRadius, owner.Random), doOrbit);
-            if (initLaunch && ship != null && !ship.IsPlatformOrStation)
+            if (initLaunch && ship != null && !ship.IsPlatformOrStation && !(ship.MaxFTLSpeed == 0 || ship.MaxSTLSpeed == 0))
```
Comment (gkapulis) [inline Ship_Game/Ships/Ship_Initialize.cs:333]:
```suggestion
            if (initLaunch && ship != null && !ship.IsPlatformOrStation && (ship.MaxFTLSpeed > 0 || ship.MaxSTLSpeed > 0))

```

File: Ship_Game/Ships/Ship_Initialize.cs [hunk 4]
```diff
@@ -330,9 +330,13 @@ public static Ship CreateShipNearPlanet(UniverseState us, string shipName, Empir
         {
             float randomRadius = owner.Random.Float(p.Radius - 100, p.Radius + 100);
             Ship ship = CreateShipAt(us, shipName, owner, p, p.Position.GenerateRandomPointOnCircle(randomRadius, owner.Random), doOrbit);
-            if (initLaunch && ship != null && !ship.IsPlatformOrStation)
+            if (initLaunch && ship != null && !ship.IsPlatformOrStation && !(ship.MaxFTLSpeed == 0 || ship.MaxSTLSpeed == 0))
                 ship.InitLaunch(LaunchPlan.Planet, p);
-
+            else if (initLaunch && ship != null && (ship.MaxFTLSpeed == 0 || ship.MaxSTLSpeed == 0))
+            {
+                ship.Position += 2;
```
Comment (gkapulis) [inline Ship_Game/Ships/Ship_Initialize.cs:337]: What is this for ?

Comment (SmischenkoB) [reply Ship_Game/Ships/Ship_Initialize.cs:338]: to make starting position not straight above planet

Comment (gkapulis) [reply Ship_Game/Ships/Ship_Initialize.cs:338]: you can do it like this:
```
                Ship createdStartingShip = Ship.CreateShipNearPlanet(UState, starterShip, empire, homePlanet, true);
                if (createdStartingShip != null && (createdStartingShip.MaxFTLSpeed == 0 || createdStartingShip.MaxSTLSpeed == 0))
                {
                    createdStartingShip.Position = homePlanet.Position.GenerateRandomPointOnCircle(1000 + homePlanet.Radius, UState.Random);
                    createdStartingShip.TetherToPlanet(homePlanet);

                }```

File: Ship_Game/Ships/Ship_Initialize.cs [hunk 5]
```diff
@@ -330,7 +330,7 @@ public static Ship CreateShipNearPlanet(UniverseState us, string shipName, Empir
         {
             float randomRadius = owner.Random.Float(p.Radius - 100, p.Radius + 100);
             Ship ship = CreateShipAt(us, shipName, owner, p, p.Position.GenerateRandomPointOnCircle(randomRadius, owner.Random), doOrbit);
-            if (initLaunch && ship != null && !ship.IsPlatformOrStation)
+            if (initLaunch && ship != null && !ship.IsPlatformOrStation && (ship.MaxFTLSpeed > 0 || ship.MaxSTLSpeed > 0))
```
Comment (gkapulis) [inline Ship_Game/Ships/Ship_Initialize.cs:333]: need to remove this part.

**Review by gkapulis — APPROVED** [review]

**Pull request merged**